tgui/packages/tgui/interfaces/NuclearBomb.js

```diff
@@ -0,0 +1,93 @@
+import { useBackend } from '../backend';
+import { Box, Button, LabeledList, Section } from '../components';
+import { Window } from '../layouts';
+
+export const NuclearBomb = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  if (!data.extended) {
+    return (
+      <Window>
+        <Window.Content>
+          <Section title="Deployment">
+            <Button
+              icon="exclamation-triangle"
+              content="Deploy Nuclear Device (will bolt device to floor)"
+              onClick={() => act("deploy")} />
+          </Section>
+        </Window.Content>
+      </Window>
+    );
+  }
+
+  return (
+    <Window>
+      <Window.Content>
+        <Section title="Authorization">
+          <LabeledList>
+            <LabeledList.Item label="Auth Disk">
+              <Button
+                icon={data.authdisk ? 'eject' : 'id-card'}
+                selected={data.authdisk}
+                content={data.diskname ? data.diskname : '-----'}
+                tooltip={data.authdisk ? "Eject Disk" : "Insert Disk"}
+                onClick={() => act("auth")} />
+            </LabeledList.Item>
+            <LabeledList.Item label="Auth Code">
+              <Button
+                icon="key"
+                disabled={!data.authdisk}
+                selected={data.authcode}
+                content={data.codemsg}
+                onClick={() => act("code")} />
+            </LabeledList.Item>
+          </LabeledList>
+        </Section>
+        <Section title="Arming & Disarming">
+          <LabeledList>
+            <LabeledList.Item label="Bolted to floor">
+              <Button
+                icon={data.anchored ? 'check' : 'times'}
+                selected={data.anchored}
+                disabled={!data.authfull}
+                content={data.anchored ? 'YES' : 'NO'}
+                onClick={() => act("toggle_anchor")} />
+            </LabeledList.Item>
+            {data.authfull && (
+              <LabeledList.Item label="Time Left">
+                <Button
+                  icon="stopwatch"
+                  content={data.time}
+                  disabled={!data.authfull}
+                  tooltip="Set Timer"
+                  onClick={() => act("set_time")} />
+              </LabeledList.Item>
+            ) || (
+              <LabeledList.Item label="Time Left"
+                color={data.timer ? "red" : ""}>
+                {data.time}
```

Comment:
```suggestion
                {data.time + 's'}
```
Should have mentioned this, but having a unit here would look nicer